rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -877,42 +821,32 @@ auto FleetUpdateHandle::Implementation::allocate_tasks(
 
   for (const auto& t : task_managers)
   {
-    states.push_back(t.second->expected_finish_state());
-    constraints_set.push_back(t.first->task_planning_constraints());
-    const auto requests = t.second->requests();
-    pending_requests.insert(
-      pending_requests.end(), requests.begin(), requests.end());
-  }
-
-  // Remove the request to be ignored if present
```

Comment:
I'd suggest keeping this original implementation that does not require the internal task queue of the task manager to be exposed. Another benefit is that it calls `TaskManager::requests()` which automatically returns requests that are not charging one. In the future, as we implement other kinds of auto included tasks such as parking, we will only need to update the implementation of `TaskManager::requests()` without modifying this block.